t/t2407-sdbus.t

```diff
@@ -0,0 +1,246 @@
+#!/bin/sh
+# ci=system
+
+test_description='Test flux systemd sd-bus bridge standalone'
+
+. $(dirname $0)/sharness.sh
+
+if ! flux version | grep systemd; then
+	skip_all="flux was not built with systemd"
+	test_done
+fi
+if ! systemctl --user show --property Version; then
+	skip_all="user systemd is not running"
+	test_done
+fi
+
+test_under_flux 1 minimal
+
+flux setattr log-stderr-level 1
+
+#
+# N.B. ListUnitsByPatterns response payload is a 'params' array whose first
+# and only item (".params[0]") is an array of units.  The jq(1) expression
+# ".params[0][][0]" extracts the first field of each unit array entry (the
+# name is field 0) and prints one entry per line.
+#
+
+# Usage: bus_list_units PATTERN
+bus_list_units() {
+    flux python -c "import flux; print(flux.Flux().rpc(\"sdbus.call\",{\"member\":\"ListUnitsByPatterns\",\"params\":[[],[\"$1\"]]}).get_str())"
+}
+
+# Usage: bus_list_units_parsed PATTERN FIELDNUM
+bus_list_units_parsed() {
+    bus_list_units "$1" >tmp.json || return 1
+    $jq -r ".params[0][][$2]" <tmp.json
+}
+
+# Usage: bus_count_units PATTERN
+bus_count_units() {
+    bus_list_units_parsed "$1" 0 >tmp.list || return 1
+    wc -l <tmp.list
+}
+
```

Comment:
I think some, possibly not all, of your tests below could be racy against these list/count checks.  i.e. you start a unit then check if its running via listing, but it may not be running by the time you check.  IIRC the kill tests can also be racy, no guarantee that the signal has been received by the process by the time you check.

Perhaps it would be safer to wrapper these in a loop and spin if the number / string / whatever isn't what you expect it to be yet.

Reply:
Ugh, I think you're right.  Let me rework.